src/test/stateful_modelinstance_test.cpp

```diff
@@ -174,6 +244,210 @@ TEST_F(StatefulModelInstanceTempDir, modelInstanceFactory) {
+    ASSERT_TRUE(status.ok());
+    auto modelInstance = manager.findModelInstance(dummyModelName);
+    tensorflow::serving::PredictResponse response;
+
+    tensorflow::serving::PredictRequest request = preparePredictRequest(modelInput);
+    setRequestSequenceControl(&request, SEQUENCE_END);
+
+    // Do the inference
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::SEQUENCE_ID_NOT_PROVIDED);
+
+    request = preparePredictRequest(modelInput);
+    setRequestSequenceControl(&request, NO_CONTROL_INPUT);
+
+    // Do the inference
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::SEQUENCE_ID_NOT_PROVIDED);
+}
+
+TEST_F(StatefulModelInstanceTempDir, statefulInferSequenceIdNotProvided2) {
+    ConstructorEnabledModelManager manager;
+    std::unique_ptr<ovms::ModelInstanceUnloadGuard> unload_guard;
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+    auto modelInstance = manager.findModelInstance(dummyModelName);
+    tensorflow::serving::PredictResponse response;
+
+    tensorflow::serving::PredictRequest request = preparePredictRequest(modelInput);
+
+    // Do the inference
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::SEQUENCE_ID_NOT_PROVIDED);
+
+    request = preparePredictRequest(modelInput);
+    setRequestSequenceControl(&request, 99);
+
+    // Do the inference
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::INVALID_SEQUENCE_CONTROL_INPUT);
+}
+
+TEST_F(StatefulModelInstanceTempDir, statefulInferIdAlreadyExists) {
+    ConstructorEnabledModelManager manager;
+    std::unique_ptr<ovms::ModelInstanceUnloadGuard> unload_guard;
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+    auto modelInstance = manager.findModelInstance(dummyModelName);
+    tensorflow::serving::PredictResponse response;
+    uint64_t seqId = 1;
+
+    tensorflow::serving::PredictRequest request = preparePredictRequest(modelInput);
+    setRequestSequenceId(&request, seqId);
+    setRequestSequenceControl(&request, SEQUENCE_START);
+
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::OK);
+    ASSERT_EQ(modelInstance->infer(&request, &response, unload_guard), ovms::StatusCode::SEQUENCE_ALREADY_EXISTS);
+}
+
+TEST_F(StatefulModelInstanceTempDir, statefulInferStandardFlow) {
+    ConstructorEnabledModelManager manager;
+    std::unique_ptr<ovms::ModelInstanceUnloadGuard> unload_guard;
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+    auto modelInstance = manager.findModelInstance(dummyModelName);
+    tensorflow::serving::PredictResponse response;
```

Comment:
Instead of `request, request2, request3` and `response, response2, response3` could we go with
`firstRequest, intermediateRequest, lastRequest` and `firstResponse, intermediateResponse, lastResponse`?